src/contexts/partner/PartnerContext.tsx

```diff
@@ -0,0 +1,164 @@
+import { captureException } from '@sentry/nextjs'
+import { useRouter, useSearchParams } from 'next/navigation'
+import {
+  createContext,
+  type PropsWithChildren,
+  type ReactNode,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react'
+import SuccessMessage from './_components/SuccessMessage'
+
+type AlertToDisplay = {
+  type: AlertType
+  content: ReactNode
+}
+
+type PartnerContextType = {
+  alertToDisplay?: AlertToDisplay
+  redirectUrl: string
+}
+
+export const PartnerContext = createContext<PartnerContextType>({
+  alertToDisplay: undefined,
+  redirectUrl: '',
+})
+
+export function PartnerProvider({ children }: PropsWithChildren) {
+  const [alertToDisplay, setAlertToDisplay] = useState<
+    AlertToDisplay | undefined
+  >(undefined)
+  const [redirectUrl, setRedirectUrl] = useState('')
+
+  const searchParams = useSearchParams()
+
+  const { progression, situation } = useCurrentSimulation()
+
+  const { exportSituationAsync } = useExportSituation()
+
+  const router = useRouter()
+
+  const partnerParams: Record<string, string> | undefined = useMemo(() => {
+    const params =
+      getPartnerFromStorage() ??
+      Object.fromEntries(
+        searchParams
+          .entries()
+          .filter(([key]) => key === PARTNER_KEY || key.startsWith(PARTNER_KEY))
+      )
+
+    return Object.keys(params).length ? params : undefined
+  }, [searchParams])
+
+  const isPartnerVerified = useVerifyPartner(partnerParams?.partner)
+
+  const hasNoPartnerParam = Object.keys(partnerParams || {}).length === 0
+
+  const handleExportSituation = useCallback(async () => {
+    if (!partnerParams) return
+
+    try {
+      const { redirectUrl: redirectUrlFromResponse } =
+        (await exportSituationAsync({
+          situation,
+          partner: partnerParams[PARTNER_KEY],
+          partnerParams,
+        })) ?? {}
+
+      if (!redirectUrlFromResponse) throw new Error('No redirect URL')
+
+      setRedirectUrl(redirectUrlFromResponse)
+
+      setAlertToDisplay({
+        type: 'success',
+        content: <SuccessMessage redirectUrl={redirectUrlFromResponse} />,
+      })
+    } catch (error) {
+      captureException(error)
+
+      setAlertToDisplay({
+        type: 'error',
+        content: (
+          <div className="xs:text-left text-center">
+            <span className="mb-2 block">
+              <Emoji className="mr-2 text-base">😫</Emoji>
+              <Trans>
+                Oups ! Une erreur s'est produite au moment d'envoyer vos
+                résultats à notre partenaire. Veuillez recommencer le parcours
+                depuis l'interface du site partenaire.
+              </Trans>
+            </span>
+          </div>
+        ),
+      })
+    } finally {
+      removePartnerFromStorage()
+    }
+    // False positive
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [exportSituationAsync, situation, partnerParams, router])
```

Comment:
Why do you need to watch `exportSituationAsync` ??

Reply:
Mon linter me le demande ; ça m'a permis de remarque que je pouvais virer le commentaire au-dessus d'ailleurs